lib/discovery.js

```diff
@@ -124,12 +124,12 @@ function mixinDiscovery(DB2, db2) {
    * @param {Function} [cb] The callback function
    */
   DB2.prototype.discoverDatabaseSchemas = function(cb) {
+    var self = this;
     var options = {};
-    // if (!cb && typeof options === 'function') {
-    //   cb = options;
-    //   options = {};
-    // }
-    // options = options || {};
+    if (self.isDB2z) {
+      return cb(Error('Function not supported'));
```

Comment:
The callback should be wrapped into `process.nextTick`. See http://stackoverflow.com/questions/13085155/nodejs-process-nexttick-vs-instant-callbacking